test/e2e/vsphere_volume_placement.go

```diff
@@ -0,0 +1,225 @@
+		}
+		if len(volumePath) > 0 {
+			vsp, err := vsphere.GetVSphere()
+			Expect(err).NotTo(HaveOccurred())
+			vsp.DeleteVolume(volumePath)
+		}
+	})
+
+	/*
+		Test Steps
+		-----------
+		1. Find node with the status available and ready for scheduling.
+		2. Add label to the node. - (vsphere_e2e_label: Random UUID)
+		3. Create VMDK volume
+		4. Create POD Spec with volume path of the vmdk and NodeSelector set to label assigned in the Step 2.
+		5. Create POD and wait for POD to become ready.
+		6. Verify volume is attached to the node labeled in the Step 2.
+		7. Delete POD.
+		8. Wait for volume to be detached from the node.
+		9. Repeat Step 5 to 7 and make sure back to back pod creation with the same volume is working as expected.
+		10. Delete VMDK volume
+	*/
+	framework.KubeDescribe("Test Back-to-back pod creation/deletion with the same volume source on the same worker node", func() {
+		It("should provision pod on the node with matching label", func() {
+			pod := createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, volumePath)
+			deletePodAndWaitForVolumeToDetach(c, ns, vsp, node1Name, pod, volumePath)
+
+			By("Creating pod on the same node: " + node1Name)
+			pod = createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, volumePath)
+			deletePodAndWaitForVolumeToDetach(c, ns, vsp, node1Name, pod, volumePath)
+		})
+	})
+
+	/*
+		Test Steps
+		-----------
+		1. Find two nodes with the status available and ready for scheduling.
+		2. Add labels to the both nodes. - (vsphere_e2e_label: Random UUID)
+		3. Create VMDK volume
+		4. Create POD Spec with volume path of the vmdk and NodeSelector set to node1's label.
+		5. Create POD and wait for POD to become ready.
+		6. Verify volume is attached to the node.
+		7. Delete POD.
+		8. Wait for volume to be detached from the node1.
+		9. Create POD Spec with volume path of the vmdk and NodeSelector set to node2's label.
+		10. Create POD and wait for POD to become ready.
+		11. Verify volume is attached to the node2.
+		12. Delete POD.
+		13. Delete VMDK volume.
+	*/
+	framework.KubeDescribe("Test Back-to-back pod creation/deletion with the same volume source attach/detach to different worker nodes", func() {
+		It("should provision pod on the node with matching label", func() {
+			pod := createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, volumePath)
+			deletePodAndWaitForVolumeToDetach(c, ns, vsp, node1Name, pod, volumePath)
+
+			By("Creating pod on the another node: " + node2Name)
+			pod = createPodWithVolumeAndNodeSelector(c, ns, vsp, node2Name, node2KeyValueLabel, volumePath)
+			deletePodAndWaitForVolumeToDetach(c, ns, vsp, node2Name, pod, volumePath)
+		})
+	})
+})
+
+func createPodWithVolumeAndNodeSelector(client clientset.Interface, namespace string, vsp *vsphere.VSphere, nodeName string, nodeKeyValueLabel map[string]string, volumePath string) (pod *v1.Pod) {
```

Comment:
Done. Removed parenthesis and named return for singe return func. removed named return from all other func where returning values are less then 3. Thank you for thorough review.